### ISSUE TITLE
Docker cache-from in push_latest workflow not referring to correct image

### ISSUE THREAD

**Issue by maresb** [mamba-org/micromamba-docker#281]
The [cache-from](https://github.com/mamba-org/micromamba-docker/blob/42792164625231845f06279a8849c49017edb110/.github/workflows/push_latest.yml#L128) for the docker build step refers to `mambaorg/micromamba:latest`. Thus only the default base image of `debian:bullseye-slim` has a usable cache. As a consequence, the non-default images are rebuilt from scratch due to commits to `main` which are irrelevant to the image.

I think that `cache-from` should be set to something like `${{ steps.set_image_variables.outputs.tag }}`, but I'm not sure.

**Comment by wholtz** [comment]
Good catch. Fixed in #283. Where I also pull the cache from ghrc in hopes that it is faster than dockerhub.

**Comment by wholtz** [comment]
I have looked into this a little. Still need to spend more time on it. The one thought I had is that the debian and ubuntu base images get updated about once a month (some times more frequently if there are critical security updates).  Those updates should break the cache. But I just looked through a bunch of the image digests and it does appear that every PR is resulting in a new digest.

**Comment by maresb** [comment]
Ya, I think ideally we would explicitly specify the sha256 of the debian/ubuntu images in the Dockerfile. That would have the added benefit of making cache-breaking explicit. We might be able to repurpose some portion of my aforementioned Python script. (Feel free to copy from it if you'd like.)

**Comment by wholtz** [comment]
It appears that caching of the first stage is now working, but for some reason the second stage thinks the cache is completely invalid. Not sure why.

**Comment by wholtz** [comment]
https://github.com/moby/buildkit/issues/2822
